### PR TITLE
Chore/upgrade to richie 2.8.1

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

We want to upgrade all sites to richie v2.8.1

## Proposal

- [x] Run `bin/upgrade 2.8.1 -c` and update CHANGELOG.md
- [x] Upgrade to Django CMS 3.9.0 
- [x] Upgrade frontend dependencies
- [x] Refactor xiti code to build upon https://github.com/openfun/richie/pull/1399

### CODE REVIEW

File: sites/funmooc/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Changed
 
+- Remove `MARKETING_CONTEXT` and use `WEB_ANALYTICS_*` context instead
```
Comment (sampaccoud) [inline sites/funmooc/CHANGELOG.md:17]: In commit message:
`Richie provides now a web_analytics context > you mean "context variable"

So we do need anymore to the marketing_context > we do not need to add it to `marking_context` anymore

File: sites/funmooc/src/backend/base/context_processors.py [hunk 2]
```diff
@@ -3,48 +3,23 @@
...
-            .distinct()
-            .values_list("code", flat=True)
-        )
-
     def get_privacy_policy_uri():
         """
         Check if a privacy policy page is published in the request language
         then return its url.
         """
+        language = get_language_from_request(request, check_path=True)
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/base/context_processors.py:21]: why not keep it outside of this function?

**Review by sampaccoud — APPROVED** [review]
Wow amazing :astonished:  :muscle:

File: sites/funmooc/src/backend/base/context_processors.py (hunk 2 above)
Comment (jbpenrath) [reply sites/funmooc/src/backend/base/context_processors.py:21]: `language` is now used only in one context so I move this variable near to the context which uses it. 🤷‍♂️

**Pull request merged**